hnn_core/network.py

```diff
@@ -1206,7 +1211,7 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
             gid_pairs[src_gid] = target_src_pair
 
         conn['src_type'] = self.gid_to_type(src_gids[0])
-        conn['src_gids'] = list(set(src_gids))
+        conn['src_gids'] = set(src_gids)
```

Comment:
Any reason to prefer sets over lists here? My main rationale for using lists was that the API for accessing them is much simpler (see the DOC string problems), and I think in general users are more comfortable with interacting with lists

Reply:
Also saw your comment below. I totally follow/agree that using sets internally is nice, but for accessing these variables after the fact I might prefer converting to list

Reply:
My main goal was to make explicit where order matters and where it doesn't. Ambiguity here is what made the bug addressed by this PR so tricky to diagnose. By using a `set` instead of `list`, we force users (and developers) to interact with the source/target gids as though they are unordered collections of unique elements, which mirrors how they are used in `NetworkBuilder`. `set` objects also have some nice methods for set comparison, element sampling, etc. that users may find helpful.

Reply:
Gotcha that's convincing enough for me!